src/main/java/org/folio/consortia/service/impl/SharingInstanceServiceImpl.java

```diff
@@ -44,13 +57,50 @@ public SharingInstance getById(UUID consortiumId, UUID actionId) {
   @Override
   @Transactional
   public SharingInstance start(UUID consortiumId, SharingInstance sharingInstance) {
-    log.debug("start:: Trying to start instance sharing: {} for consortium: {}", sharingInstance.getInstanceIdentifier(), consortiumId);
+    log.debug("start:: Trying to start instance sharing with instanceIdentifier: {}, consortiumId: {}.", sharingInstance.getInstanceIdentifier(), consortiumId);
     consortiumService.checkConsortiumExistsOrThrow(consortiumId);
-    checkTenantsExistAndContainCentralTenantOrThrow(sharingInstance.getSourceTenantId(), sharingInstance.getTargetTenantId());
 
-    SharingInstanceEntity savedSharingInstance = sharingInstanceRepository.save(toEntity(sharingInstance));
-    log.info("start:: sharingInstance with id: {}, instanceId: {}, sourceTenantId: {}, targetTenantId: {} successfully saved in db",
-      savedSharingInstance.getId(), savedSharingInstance.getInstanceId(), savedSharingInstance.getSourceTenantId(), savedSharingInstance.getTargetTenantId());
+    String centralTenantId = tenantService.getCentralTenantId();
+    String sourceTenantId = sharingInstance.getSourceTenantId();
+    String targetTenantId = sharingInstance.getTargetTenantId();
+    checkTenantsExistAndContainCentralTenantOrThrow(sourceTenantId, targetTenantId);
+
+    SharingInstanceEntity savedSharingInstance;
+    if (Objects.equals(centralTenantId, sourceTenantId)) {
+      FolioExecutionContext currentTenantContext = (FolioExecutionContext) folioExecutionContext.getInstance();
```

Comment:
what is the purpose of this variable? Doesn't the code below work with a plain folioExecutionContext field?

Reply:
I try to stick general approach. Please see following files:
1. https://github.com/folio-org/mod-consortia/blob/bf65437b99771a7635bb487295492778b3217352/src/main/java/org/folio/consortia/service/impl/UserTenantServiceImpl.java#L112
2. https://github.com/folio-org/mod-consortia/blob/bf65437b99771a7635bb487295492778b3217352/src/main/java/org/folio/consortia/service/impl/UserAffiliationServiceImpl.java#L44
3. https://github.com/folio-org/mod-consortia/blob/bf65437b99771a7635bb487295492778b3217352/src/main/java/org/folio/consortia/service/impl/TenantServiceImpl.java#L92

Reply:
That's not some general approach, looks just like added one time and copypasted :)